### PR TITLE
src/handlers.rs: explicitly handle INVALIDATE, SHUTDOWN, GETFD*

### PR DESCRIPTION
**Pull request by flokli**
In both cases, we want to no-op. The client doesn't expect any response from nscd, but let's at least log a debug message on why we're no-op'ing here, to distinguish from the operations that are not yet implemented.

### CODE REVIEW

**Comment by flokli** [comment]
I saw the `GETFD*` requests on the wire when working on reverse host lookups (while adding `GETHOSTBYADDR*)` support as part of https://github.com/twosigma/nsncd/issues/37.

@geofft @leifwalsh WDYT?

**Comment by leifwalsh** [comment]
Seems reasonable to me

**Review by geofft — APPROVED** [review]
Hmm, not unreasonable but I'm curious about the motivation. I feel like I'd do it the other way around - all of these really should get no response but the rest of them are unimplemented and it might be worth noting if we get one.

Should we (also? instead?) log the unimplemented requests? The enum derives Debug so we can just log the actual request value.

(Also now that I'm looking at this it occurs to me that `LASTREQ` isn't actually a valid protocol request. :) )

File: src/handlers.rs
Comment (geofft) [inline src/handlers.rs:145]: This returns statistics about the daemon - you can also mark this as will-not-implement.

Comment (flokli) [reply src/handlers.rs:145]: Why? I've been thinking about exposing some prometheus-style metrics on the number of requests, success rate and timing Information, by request type. That'd of course primarily be exposed via http, but if that getstat call is actually usable to expose some of these numbers, and there's some userspace tool to query and show, why not?

**Comment by flokli** [comment]
> Hmm, not unreasonable but I'm curious about the motivation. I feel like I'd do it the other way around - all of these really should get no response but the rest of them are unimplemented and it might be worth noting if we get one.

I was mostly concerned about having a distinction in code if something has not been implemented yet, or if we consciously decided it's okay to return no data, and userspace can cope with it.

> 
> Should we (also? instead?) log the unimplemented requests? The enum derives Debug so we can just log the actual request value.

I think we already `debug!` the request in `main.rs`.

> (Also now that I'm looking at this it occurs to me that `LASTREQ` isn't actually a valid protocol request. :) )

Can you elaborate? It's not something that ever happens to be sent out by the client, or it's something where we should always pretend to have not yet have had any previous request?

**Comment by flokli** [comment]
Btw, can these workflows be changed to not require approval? It'd be nice to immediately get feedback on the various GH action checks...

**Comment by leifwalsh** [comment]
It's set to require approval for first time contributors, to prevent something like someone running a crypto miner using actions. Once we get you that CLA (#44) and merge some PRs from you you'll be good.

**Comment by flokli** [comment]
Is this good to go?

**Comment by leifwalsh** [comment]
> I feel like I'd do it the other way around - all of these really should get no response but the rest of them are unimplemented and it might be worth noting if we get one.
> 
> Should we (also? instead?) log the unimplemented requests? The enum derives Debug so we can just log the actual request value.

I like the gist of this: it clarifies which operations we'll never implement and makes it clear which remain to be implemented. We can log the unimplemented ones (at info? warn?) later.

I'm gonna merge this for now, we can tweak later (e.g. doing _something_ with `LASTREQ`).

**Pull request merged**
